aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/CreateHandler.java

```diff
@@ -0,0 +1,54 @@
+package software.amazon.rds.optiongroup;
+
+import java.util.Optional;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.OptionGroupAlreadyExistsException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    public static final int PHYSICAL_RESOURCE_ID_MAX_LENGTH = 255;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<RdsClient> proxyClient,
+        final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+        if (StringUtils.isNullOrEmpty(model.getOptionGroupName())) {
+            model.setOptionGroupName(IdentifierUtils.generateResourceIdentifier(
+                    Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse("optiongroup"),
+                    request.getClientRequestToken(),
+                    PHYSICAL_RESOURCE_ID_MAX_LENGTH
+            ).toLowerCase());
+        }
```

Comment:
TODO: move this to ProgressEvent.progress().then(), otherwise a new name would be generated every time create request fails, which is fairly dangerous.